ding_dibs.module

```diff
@@ -420,3 +442,155 @@ function ding_dibs_cards() {
 
   return $cards;
 }
+
+/**
+ * Implements hook_cron().
+ * This is done by sending written reports and email adresses result cron launches in the database
+ */
+function ding_dibs_cron() {
+  $start_cron = FALSE;
+  $start = variable_get('ding_dibs_start_cron_hour', '');
+  $start_minute = variable_get('ding_dibs_start_cron_minute', 0);
+  $start_operation = variable_get('ding_dibs_start_operation_cron', '');
+  $emails_list = variable_get('ding_dibs_emails', '');
+  if (!empty($start)) {
+    $start_date = mktime($start, $start_minute, 0, format_date(REQUEST_TIME, 'custom', 'm', date_default_timezone_get()), format_date(REQUEST_TIME, 'custom', 'd', date_default_timezone_get()), format_date(REQUEST_TIME, 'custom', 'Y', date_default_timezone_get()));
+    if (!empty($start_operation)) {
+      $start_operation_date = format_date($start_operation, 'custom', 'Y-d-m', date_default_timezone_get());
+    }
+    if (REQUEST_TIME >= $start_date) {
+      if (empty($start_operation_date) || $start_operation_date < format_date(REQUEST_TIME, 'custom', 'Y-d-m', date_default_timezone_get())) {
+        $start_cron = TRUE;
+      }
+    }
+  }
```

Comment:
This is still plagued by poor naming of variables, relying on strings for comparisons when we are working with dates and times etc.

Consider this alternate implementation of the same logic:

```php
  $start_cron = FALSE;

  $cron_start_hour = variable_get('ding_dibs_start_cron_hour', NULL);
  $cron_start_minute = variable_get('ding_dibs_start_cron_minute', 0);
  // Only start the cron if we have a valid hourly schedule.
  if (!empty($cron_start_hour)) {
    $request_time = new DateTime(REQUEST_TIME);

    $today_cron_start_time = new DateTime(REQUEST_TIME);
    $today_cron_start_time->setTime($cron_start_hour, $cron_start_minute);

    // If our cron has not run previously then use 0 as the timestamp.
    // This should ensure that DateTime(REQUEST_TIME) > DateTime(0).
    $last_cron_time = new DateTime(variable_get('ding_dibs_last_cron', 0));

    // We run our cron if it has not already run for today and we are past the
    // time when the report should be generated for today.
    if ($today_cron_start_time > $last_cron_time
      && $request_time >= $today_cron_start_time) {
        $start_cron = TRUE;
    }
  }
```

There is ness nesting, less if-statements, Drupal variables have been renamed and follow similar patterns as other modules (`ding_dibs_start_operation_cron` vs. `ctools_last_cron`, `node_cron_last`), proper DateTime objects are used for manipulation and comparison, long lines are avoided etc.